config_gen.py

```diff
@@ -136,7 +136,10 @@ def main():
             else:
                 lang, flags = ("c++", cxx_flags)
 
-            generate_conf(["-x", lang] + flags, config_file)
+            if force_lang:
+                lang += ["-x", force_lang]
+                
+            generate_conf(flags, config_file)
```

Comment:
This change will break color-coded config generation (`generate_cc_conf()`), because it outputs a plain text file without the logic in `template.py`.

IMO, a better way to handle this would be to add separate parameters to the `generate_*_conf` functions for `force_lang` and `detected_lang`, so they can do the right thing for their outputs.